package.json

```diff
@@ -53,20 +53,22 @@
         "grunt-contrib-watch": "1.1.0",
         "grunt-modulefiles": "0.4.0",
         "grunt-shell": "2.1.0",
+        "hypher": "0.2.5",
```

Comment:
When you get back from your leave we should start thinking about modularization of Infusion quite urgently since these are far too many dependencies for those who just to get access to the core framework

Reply:
Yes, we should look more into modularization. Even being able to break up the prefs framework into smaller modules would be useful. We should try to sketch out what we want for this again, or add to what we already have if there are more/new requirements. 

Regarding the above dependencies, they are all listed as dev dependencies so they aren't pulled in with NPM installs into other projects. Also, they can be omitted from various builds of infusion via the custom builds. For example, hypher wouldn't be included in the "framework" only builds.

Reply:
I can understand the reasoning for listing them as dev dependencies, but it is essentially misleading since they are expected to be used by the implementation. For example, in use cases such as chart authoring - https://github.com/fluid-project/chartAuthoring/blob/master/package.json#L11 - Infusion is consumed as a client-side dependency. Whilst the consumer could just bind to "dist", they might want to bind at the source level for debugging purposes. In that case they'd have to replicate the setup and package.json entries for hypher etc. Whilst this is an edge use case, it is part of the set of issues we'll need to consider when modularising.

Reply:
The lib files are actually included in the NPM package ( https://unpkg.com/infusion@3.0.0-dev.20181120T213100Z.8ef92ceac.FLUID-6240/src/lib/ ). They are just ignored from git. This, for example, is how we get access to the language pattern files in UIO+.